src/Oriflame.RedisMessaging.ReliableDelivery/Subscribe/MessageHandler.cs

```diff
@@ -28,42 +33,100 @@ public class MessageHandler : IMessageHandler
             _log = log ?? NullLogger<MessageHandler>.Instance;
         }
 
-        protected virtual void OnMissingMessages(Message currentMessage, long lastProcessedMessageId)
+        public void HandleMessage(Message message)
         {
-            var lostMessages = _messageLoader.GetMessages(Channel, lastProcessedMessageId + 1, currentMessage.Id - 1);
-            foreach (var lostMessage in lostMessages)
+            lock (_lock)
             {
-                OnSuccessfulMessage(lostMessage);
+                HandleMessageImpl(message);
+                LastActivityAt = Now;
             }
-            OnSuccessfulMessage(currentMessage);
         }
 
-        protected virtual void OnSuccessfulMessage(Message message)
+        protected virtual DateTime Now => DateTime.Now;
+
+        public void CheckMissedMessages()
         {
-            _onSuccessMessage(message);
+            _log.LogDebug("Checking missed messages");
+            var messages = GetNewestMessages();
+            int messagesCount = 0;
+            long firstMessageId = 0;
+            long lastMessageId = 0;
+            lock (_lock)
+            {
+                foreach (var message in messages)
+                {
+                    if (messagesCount == 0)
+                    {
+                        firstMessageId = message.Id;
+                    }
+                    HandleMessageImpl(message);
+                    ++messagesCount;
+                    lastMessageId = message.Id;
+                }
+
+                LastActivityAt = Now;
+            }
+            if (messagesCount > 0)
+            {
+                _log.LogWarning("Missed messages processed: messagesCount={messagesCount}, IDs range=<{firstMessageId}, {lastMessageId}>",
+                    messagesCount, firstMessageId, lastMessageId);
+            }
+            else
+            {
+                _log.LogInformation("Checked missed messages: no messages missed found.");

```

Comment:
LogDebug to prevent log flooding on live env...

Reply:
Changed. We should have some kind of "tracking" code that would help us to proove that a method  `MessageHandler.CheckMissedMessages()` is being called regularly, e.g. simple counter might work here.